### PR TITLE
a8n: Decouple changeset and repo syncing

### PR DESCRIPTION
**Pull request by ryanslade**
This is just a small change to decouple the syncers before we begin work
on making the changeset syncing more intelligent.

### CODE REVIEW

File: cmd/repo-updater/shared/main.go [hunk 1]
```diff
@@ -31,7 +31,9 @@ import (
 
 const port = "3182"
 
-func Main(newPreSync repos.NewPreSync, dbInitHook func(db *sql.DB)) {
+type DependencyCallback func(db *sql.DB, store repos.Store, cf *httpcli.Factory)
```
Comment (tsenart) [inline cmd/repo-updater/shared/main.go]: This name doesn't communicate the intent of this callback (i.e. it only conveys what is does, not why). Can you think of a name that does?

File: enterprise/cmd/repo-updater/main.go [hunk 2]
```diff
@@ -20,30 +21,41 @@ func main() {
...
-	newPreSync := func(db *sql.DB, rs repos.Store, cf *httpcli.Factory) func(context.Context) error {
-		syncer := &a8n.ChangesetSyncer{
-			Store:       a8n.NewStore(db),
-			ReposStore:  rs,
-			HTTPFactory: cf,
-		}
-
-		return syncer.Sync
-	}
+var cbOnce sync.Once
```
Comment (tsenart) [inline enterprise/cmd/repo-updater/main.go]: Why is a `sync.Once` needed here? It isn't obvious to me without additional context.

Comment (ryanslade) [reply enterprise/cmd/repo-updater/main.go]: Just to guard against the callback function being called more than once in `shared.Main`. Each call will kick off background processing and it should only do so once.

File: cmd/repo-updater/shared/main.go (hunk 1 above)
Comment (ryanslade) [reply cmd/repo-updater/shared/main.go]: I'm struggling to think of a better name but I'll add a comment to explain the intent

File: enterprise/cmd/repo-updater/main.go [hunk 3]
```diff
@@ -20,30 +21,41 @@ func main() {
...
-		}
-	}
-
-	shared.Main(newPreSync, dbInitHook)
+			for {
+				err := syncer.Sync(ctx)
+				if err != nil {
+					log15.Error("Syncing Changesets", "err", err)
+				}
+				time.Sleep(1 * time.Minute)
```
Comment (mrnugget) [inline enterprise/cmd/repo-updater/main.go]: Hmm, 1min is a bit much, I think. How about 2 or 3?

Comment (ryanslade) [reply enterprise/cmd/repo-updater/main.go]: I just went with that since it's what we're currently doing

Comment (mrnugget) [reply enterprise/cmd/repo-updater/main.go]: Then this might be the chance to increase it ;)

Comment (ryanslade) [reply enterprise/cmd/repo-updater/main.go]: I increased it to 2 minutes. There will be many more changes to this soon once we start doing heuristic / priority based syncing.

File: cmd/repo-updater/shared/main.go (hunk 1 above)
Comment (tsenart) [reply cmd/repo-updater/shared/main.go]: How about EnterpriseCallback? There’d be no actual enterprise code in the OSS version.

Comment (tsenart) [reply cmd/repo-updater/shared/main.go]: Or better, EnterpriseInit

**Pull request merged**